lib/manager/composer/utils.ts

```diff
@@ -73,13 +79,14 @@ export function extractContraints(
   }
   // check last used composer version
   else if (lockParsed?.['plugin-api-version']) {
-    const major = api.getMajor(lockParsed?.['plugin-api-version']);
-    res.composer = `${major}.*`;
+    res.composer = lockParsed?.['plugin-api-version'];
```

Comment:
This should be ok to pass through as it's coming from
https://github.com/composer/composer/blob/2.2.0-RC1/src/Composer/Package/Locker.php#L392
https://github.com/composer/composer/blob/2.2.0-RC1/src/Composer/Plugin/PluginInterface.php#L35

Reply:
This version isn't equal to the Composer version! So i think this is a wrong assumption.

Reply:
Yeah you're right, I wasn't even questioning this..
Same thing for `composer-runtime-api` too I would say hmm, see https://getcomposer.org/doc/articles/composer-platform-dependencies.md#runtime-package-composer-runtime-api

This is bad, as I would have to explicitly require a composer version in my projects in order to benefit from the changes introduced here.

Do you have any other ideas? I just can think of the already mentioned parsing of `composer version` so far.. :)

Reply:
Anyways, I guess it makes sense to remove the check of those 2 props in another PR, right?

Reply:
Sorry still thinking out loud..
On the other hand, maybe those 2 props are still fine to depend on. It might happen that they are not increased with a new Composer version, e.g. Composer 2.3 still sets `plugin-api-version` to 2.2, but that means that the Composer version that works with the 2.2 plugin API is needed. Which is most like Composer 2.2.

Reply:
OK, we should change this.

But in future it's best to direct reference a specific composer version: https://getcomposer.org/doc/articles/composer-platform-dependencies.md#composer-package-composer

instead of returning `${major}.*`, we should return `^${major}.${minor}` as version. Because this is the minimum composer version which caused the bump. So we should also do this for `composer-runtime-api`.

So can you change the code accordingly?

Reply:
Additionally to the `composer/composer` package, we should check `composer` package, which is available from `v2.2`

Reply:
Thx for checking! Sounds good, I'll adapt it later

Reply:
Created https://github.com/renovatebot/renovate/pull/13186 and https://github.com/renovatebot/renovate/pull/13187 and removed the relevant changes from here.